### PR TITLE
try again in case of a conflict when updating users

### PR DESCRIPTION
**Pull request by jsteemann**

### CODE REVIEW

**Comment by jsteemann** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/2390/

File: arangod/VocBase/Methods/Collections.cpp [hunk 1]
```diff
@@ -590,9 +617,23 @@ Result Collections::drop(TRI_vocbase_t* vocbase, LogicalCollection* coll,
...
-    um->enumerateUsers([&](auth::User& entry) -> bool {
-      return entry.removeCollection(dbname, collName);
-    });
+    int tries = 0;
+    while (true) {
+      Result r = um->enumerateUsers([&](auth::User& entry) -> bool {
+        return entry.removeCollection(dbname, collName);
+      });
+
+      if (r.ok() || !r.is(TRI_ERROR_ARANGO_CONFLICT)) {
```
Comment (graetzer) [inline arangod/VocBase/Methods/Collections.cpp]: you are not returning the error result here ?

File: arangod/VocBase/Methods/Collections.cpp [hunk 2]
```diff
@@ -286,10 +299,24 @@ Result Collections::create(TRI_vocbase_t* vocbase, std::string const& name,
...
-        });
+        // however, there may be races for updating the users account, so we try a
+        // few times in case of a conflict
+        int tries = 0;
+        while (true) {
+          Result r = um->updateUser(ExecContext::CURRENT->user(), [&](auth::User& entry) {
+            entry.grantCollection(vocbase->name(), name, auth::Level::RW);
+            return TRI_ERROR_NO_ERROR;
+          });
+          if (r.ok() || !r.is(TRI_ERROR_ARANGO_CONFLICT)) {
```
Comment (graetzer) [inline arangod/VocBase/Methods/Collections.cpp]: you are not returning the error in `r`

File: arangod/VocBase/Methods/Collections.cpp [hunk 3]
```diff
@@ -267,10 +266,24 @@ Result Collections::create(TRI_vocbase_t* vocbase, std::string const& name,
...
```
Comment (graetzer) [inline arangod/VocBase/Methods/Collections.cpp]: you are not returning the error

**Comment by jsteemann** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/2396/

**Comment by jsteemann** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/2411/

**Comment by not-sungard** [comment]
This sounds promising, thanks to all who worked on it.
Is there a rough date when 3.4.1 would likely see the light of day?

**Comment by jsteemann** [comment]
@not-sungard : I cannot provide a release date for 3.4.1 yet. 
First there will be the GA release of 3.4.0, but that should be real soon already.
For 3.4.1 to come out I expect at most a few weeks to pass, but that is my personal guess right now.

**Comment by not-sungard** [comment]
Ok, thanks for the info. Much appreciated.

**Pull request merged**